### PR TITLE
UnderTheSkin implemented

### PR DESCRIPTION
**Pull request by alma555**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+}
+
+class UnderTheSkinEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterPermanentCard("permanent card from your graveyard");
+
+    UnderTheSkinEffect() {
+        super(Outcome.Benefit);
+        staticText = "You may return a permanent card from your graveyard to your hand.";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:51]: add `this.concatBy("<br>");` so there's a line break

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    private UnderTheSkinEffect(final mage.cards.u.UnderTheSkinEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public mage.cards.u.UnderTheSkinEffect copy() {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:59]: make sure to remove `mage.cards.u.`

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        return new mage.cards.u.UnderTheSkinEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:64]: the apply method here is perfectly fine so everything after this is just nitpicking

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        TargetCard target = new TargetCardInYourGraveyard(0, 1, filter, true);
+        if (!player.choose(Outcome.ReturnToHand, target, source, game)) {
+            return true;
+        }
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:72]: you can just call the choose method without it being an if statement

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        Card card = game.getCard(target.getFirstTarget());
+        if (card == null) {
+            return true;
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:75]: this should be false

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        }
+        player.moveCards(card, Zone.HAND, source, game);
+        return true;
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/u/UnderTheSkin.java:78]: you can actually simplify the whole thing into `return card != null && player.moveCards(card, Zone.HAND, source, game);`

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java (hunk 4 above)
Comment (alma555) [reply Mage.Sets/src/mage/cards/u/UnderTheSkin.java:72]: I copied from your implementation of Acolyte of Affliction, which uses the same formatting. It's not really a pressing issue, but Is there a reason why you used it there?

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java (hunk 6 above)
Comment (alma555) [reply Mage.Sets/src/mage/cards/u/UnderTheSkin.java:78]: Oh, that's neat. I feel like having two distinct checks is more readable though.

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java (hunk 4 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/u/UnderTheSkin.java]: yeah it's just an older card and I never went back and fixed it, but we should probably refactor this effect into a shared one anyway (outside the scope of this PR)

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java (hunk 6 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/u/UnderTheSkin.java:79]: you should at least return the result of the moveCards method, the boolean is supposed to reflect whether the game state has changed (this is not important)

File: Mage.Sets/src/mage/cards/u/UnderTheSkin.java (hunk 2 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/u/UnderTheSkin.java:60]: don't forget this one

**Review by theelk801 — CHANGES_REQUESTED** [review]

**Review by theelk801 — APPROVED** [review]
thanks for the contribution

**Pull request merged**